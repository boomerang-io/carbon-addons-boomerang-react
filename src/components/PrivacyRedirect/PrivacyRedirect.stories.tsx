import React from "react";
import PrivacyRedirect from "./PrivacyRedirect";
import { User } from "../../types";

export default {
  title: "Platform/PrivacyRedirect",
  component: PrivacyRedirect,
  parameters: {
    docs: {
      description: {
        component: "Header modal showing GDPR redirect for users that need to accept terms and conditions",
      },
    },
  },
};

export const Default = () => {
  return <PrivacyRedirect isOpen baseEnvUrl="" platformName="Boomerang" user={{ status: "active" } as User} />;
};

export const PendingDeletion = () => {
  return <PrivacyRedirect isOpen baseEnvUrl="" platformName="Boomerang" user={{ status: "pending_deletion" } as User} />;
};
