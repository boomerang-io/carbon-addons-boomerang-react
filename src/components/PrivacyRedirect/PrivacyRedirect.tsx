/*
IBM Confidential
694970X, 69497O0
© Copyright IBM Corp. 2022, 2024
*/


import React from "react";
import { ComposedModal, Button, ModalHeader, ModalBody, ModalFooter } from "@carbon/react";
import { prefix } from "../../internal/settings";
import type { User } from "types";

type Props = {
  baseEnvUrl: string;
  isOpen?: boolean;
  platformName?: string;
  user?: User;
};

function PrivacyRedirect(props: Props) {
  const { baseEnvUrl, isOpen = false, user, platformName = "the platform" } = props;

  const handleOnSubmit = () => {
    const pendingDeletion = user?.status === "pending_deletion";
    if (pendingDeletion) {
      window.location.assign(`${baseEnvUrl}/launchpad`); // There is no marketing site so TBD where this links to
    } else {
      window.location.assign(`${baseEnvUrl}/launchpad?rd=${document.URL}`);
    }
  };

  const pendingDeletion = user?.status === "pending_deletion";
  const contentText = pendingDeletion
    ? `We’re working on removing your account and personal information from ${platformName}. Please allow up to 1 month (as mandated by GDPR regulations) for us to complete your request.`
    : "Before continuing, we need you to consent to the Privacy Statement.";

  const buttonText = pendingDeletion ? "Go to Launchpad" : "View Privacy Statement";

  return (
    <div className={`${prefix}--bmrg-redirect-container ${prefix}--bmrg-header-modal`}>
      <ComposedModal
        aria-label="GDPR Modal"
        open={isOpen}
        onClose={() => {
          return false;
        }}
      >
        <ModalHeader title="Our Privacy Statement" />
        <ModalBody>
          <span className={`${prefix}--bmrg-redirect__body`}>{contentText}</span>
        </ModalBody>
        <ModalFooter>
          <Button data-modal-primary-focus kind="primary" onClick={handleOnSubmit}>
            {buttonText}
          </Button>
        </ModalFooter>
      </ComposedModal>
    </div>
  );
}

export default PrivacyRedirect;
