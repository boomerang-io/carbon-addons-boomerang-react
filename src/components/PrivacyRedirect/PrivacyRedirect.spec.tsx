import React from "react";
import { expect, test } from "vitest";
import { screen, render } from "@testing-library/react";
import { axe } from "jest-axe";
import PrivacyRedirect from "./PrivacyRedirect";

describe("PrivacyRedirect", () => {
  test("snapshot", async () => {
    const { baseElement } = render(<PrivacyRedirect isOpen baseEnvUrl="https://ibm.com" />);
    expect(baseElement).toMatchSnapshot();
  });

  test("default message", async () => {
    render(<PrivacyRedirect isOpen baseEnvUrl="https://ibm.com" />);
    expect(screen.getByText("Before continuing, we need you to consent to the Privacy Statement.")).toBeInTheDocument();
  });

  test("pending deletion message", async () => {
    render(<PrivacyRedirect isOpen baseEnvUrl="https://ibm.com" user={{ status: "pending_deletion" }} />);
    expect(screen.getByText(/Weâ€™re working on removing your account and personal information./)).toBeInTheDocument();
  });

  test("a11y", async () => {
    const { container } = render(<PrivacyRedirect isOpen baseEnvUrl="https://ibm.com" />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
