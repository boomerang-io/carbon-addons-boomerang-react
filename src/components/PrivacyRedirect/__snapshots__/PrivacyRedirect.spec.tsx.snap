// Vitest Snapshot v1

exports[`PrivacyRedirect > snapshot 1`] = `
<body
  class="cds--body--with-modal-open"
>
  <div>
    <div
      class="cds--bmrg-redirect-container cds--bmrg-header-modal"
    >
      <div
        aria-hidden="false"
        class="cds--modal is-visible"
        role="presentation"
        tabindex="-1"
      >
        <div
          aria-label="GDPR Modal"
          aria-modal="true"
          class="cds--modal-container"
          role="dialog"
        >
          <button
            class="cds--visually-hidden"
            type="button"
          >
            Focus sentinel
          </button>
          <div
            class="cds--modal-container-body"
          >
            <div
              class="cds--modal-header"
            >
              <h3
                class="cds--modal-header__heading cds--type-beta"
              >
                Our Privacy Statement
              </h3>
              <button
                aria-label="Close"
                class="cds--modal-close"
                title="Close"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="cds--modal-close__icon"
                  fill="currentColor"
                  focusable="false"
                  height="20"
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 32 32"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M24 9.4L22.6 8 16 14.6 9.4 8 8 9.4 14.6 16 8 22.6 9.4 24 16 17.4 22.6 24 24 22.6 17.4 16 24 9.4z"
                  />
                </svg>
              </button>
            </div>
            <div
              class="cds--modal-content"
            >
              <span
                class="cds--bmrg-redirect__body"
              >
                Before continuing, we need you to consent to the Privacy Statement.
              </span>
            </div>
            <div
              class="cds--modal-footer cds--btn-set"
            >
              <button
                class="cds--btn cds--btn--primary"
                tabindex="0"
                type="button"
              >
                View Privacy Statement
              </button>
            </div>
          </div>
          <button
            class="cds--visually-hidden"
            type="button"
          >
            Focus sentinel
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`PrivacyRedirect - snapshot 1`] = `
<body
  class="cds--body--with-modal-open"
>
  <div>
    <div
      class="cds--bmrg-redirect-container cds--bmrg-header-modal"
    >
      <div
        aria-hidden="false"
        class="cds--modal is-visible"
        role="presentation"
        tabindex="-1"
      >
        <div
          aria-label="GDPR Modal"
          aria-modal="true"
          class="cds--modal-container"
          role="dialog"
        >
          <button
            class="cds--visually-hidden"
            type="button"
          >
            Focus sentinel
          </button>
          <div
            class="cds--modal-container-body"
          >
            <div
              class="cds--modal-header"
            >
              <h3
                class="cds--modal-header__heading cds--type-beta"
              >
                Our Privacy Statement
              </h3>
              <button
                aria-label="Close"
                class="cds--modal-close"
                title="Close"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="cds--modal-close__icon"
                  fill="currentColor"
                  focusable="false"
                  height="20"
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 32 32"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M24 9.4L22.6 8 16 14.6 9.4 8 8 9.4 14.6 16 8 22.6 9.4 24 16 17.4 22.6 24 24 22.6 17.4 16 24 9.4z"
                  />
                </svg>
              </button>
            </div>
            <div
              class="cds--modal-content"
            >
              <span
                class="cds--bmrg-redirect__body"
              >
                Before continuing, we need you to consent to the Privacy Statement.
              </span>
            </div>
            <div
              class="cds--modal-footer cds--btn-set"
            >
              <button
                class="cds--btn cds--btn--primary"
                tabindex="0"
                type="button"
              >
                View Privacy Statement
              </button>
            </div>
          </div>
          <button
            class="cds--visually-hidden"
            type="button"
          >
            Focus sentinel
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
